src/classes/CON_PrimaryAffls_TDTM.cls

```diff
@@ -51,151 +51,200 @@ public class CON_PrimaryAffls_TDTM extends TDTM_Runnable {
+    * @description Sets the value of CON_PrimaryAffls_TDTM_After_Insert recursion flag to false
+    *******************************************************************************************************/
+    @TestVisible
+    private void unsetInsertRecursion() {
+        TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.registeredTrigger.CON_PrimaryAffls_TDTM_After_Insert, false);
+    }
+
+    /************************************************************************************************************************************
+    * @description Returns the value of operation (AFFL_AccRecordType_TDTM OR CON_PrimaryAffls_TDTM_After_Update)
+    **************************************************************************************************************************************/
+    @TestVisible
+    private boolean isUpdateRecursion() {
+        return TDTM_ProcessControl.getRecursionFlag(TDTM_ProcessControl.registeredTrigger.AFFL_AccRecordType_TDTM) || 
+            TDTM_ProcessControl.getRecursionFlag(TDTM_ProcessControl.registeredTrigger.CON_PrimaryAffls_TDTM_After_Update);
+    }
+
+    /************************************************************************************************************************************
+    * @description Sets the recursion flag CON_PrimaryAffls_TDTM_After_Update) to true
+    **************************************************************************************************************************************/
+    @TestVisible
+    private void setUpdateRecursion() {
+        TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.registeredTrigger.CON_PrimaryAffls_TDTM_After_Update, true);
+    }
+
+    /************************************************************************************************************************************
+    * @description unsets the recursion flag CON_PrimaryAffls_TDTM_After_Update) to false
+    **************************************************************************************************************************************/
+    @TestVisible
+    private void unsetUpdateRecursion() {
+        TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.registeredTrigger.CON_PrimaryAffls_TDTM_After_Update, false);
+    }
+
+    /*******************************************************************************************************
+    * @description Returns if Contact Deletion Prevention custom setting is not enabled or
+    * Passes a map of contacts By Id to SRVC_Contact_PreventDeletion and returns a List of Contacts that have
+    * been prevented for deletion
+    * @param newSObjectList list of sObject records that are being deleted
+    * @return void
+    ********************************************************************************************************/
+    @TestVisible
+    private void handleAfterInsert(List<SObject> newSObjectList) {
+
+        if (this.isInsertRecursion() == true) {
```

Comment:
Could be wrong but I thought we were doing guard clause for newlist in AfterInsert and newlist&oldlist in AfterUpdate?

Reply:
We are doing it further up the food chain in the trigger run method when we can. Even faster bailing.